src/middleware/authentication.ts

```diff
@@ -60,35 +62,26 @@ async function decodeAuthChain(req: Request): Promise<string> {
   let ethAddress: string | null = null
   let errorMessage: string | null = null
 
-  if (authChain.length === 0) {
+  if (!Authenticator.isValidAuthChain(authChain)) {
     errorMessage = `Invalid auth chain`
   } else {
-    ethAddress = authChain[0].payload
+    ethAddress = Authenticator.ownerAddress(authChain)
 
     if (!ethAddress) {
       errorMessage = 'Missing ETH address in auth chain'
     } else {
       try {
-        // TODO: We are waiting for the final implementation of https://github.com/decentraland/decentraland-crypto-middleware in order to complete use it.
-        // For the time being, we need to reduce the number of request to the catalysts
-        const endpoint = (req.method + ':' + req.path).toLowerCase()
-        if (isEIP1664AuthChain(authChain)) {
-          // We don't use the response, just want to make sure it does not blow up
-          await peerAPI.validateSignature({ authChain, timestamp: endpoint }) // We send the endpoint as the timestamp, yes
-        } else {
-          const res = await Authenticator.validateSignature(
-            endpoint,
-            authChain,
-            null as any,
-            Date.now()
-          )
-
-          if (!res.ok) {
-            errorMessage = res.message!
-          }
-        }
+        await verify(req.method, req.path, req.headers, {
+          fetcher: peerAPI.signatureFetcher
+        })
       } catch (error) {
         errorMessage = isErrorWithMessage(error) ? error.message : 'Unknown'
+        try {
+          await validateSignature(req, authChain)
+          errorMessage = null // clear error if it has success
+        } catch (_) {
+          // we use the error reported 
```

Comment:
Should we override the `errorMessage` variable here? The `validateSignature` throws an error that might be different from the one that is thrown by the `verify` function
```suggestion
      catch (error) {
           errorMessage = isErrorWithMessage(error) ? error.message : 'Unknown'
       }
```